- Added movie search logic.

Kevin-Guillen opened the pull request:
Added the movie search logic , reusing the swiper to display the search or trending movies depending on if the textfield is empty or not. The title above the swiper will change dynamically.



https://user-images.githubusercontent.com/84336422/125364271-8c8af700-e348-11eb-9b26-abed4e6247a8.mp4

Kevin-Guillen committed:
- Added movie search logic.

FranciscoGaraventa commented on lib/src/ui/movies_page.dart:
```diff
@@ -118,6 +123,7 @@ class _MoviesPageState extends State<MoviesPage> {
             children: <Widget>[
               CardSwiper(
                 trendingMovies: _trendingMovies,
+                bloc: _moviesBloc,
```
Is possible to send only the boolean instead of the complete bloc?. And please rename the parameter name is not descriptive.

Kevin-Guillen replied:
done.

Kevin-Guillen committed:
- Comments fixed:
- Send only "isTextField" instead of the complete bloc
- Renamed parameters.
- Added controls to the swiper.
- Added missing commas
- Added messages to the constants file

FranciscoGaraventa commented on lib/src/widgets/card_swiper.dart:
```diff
@@ -4,9 +4,11 @@ import '../model/movies.dart';
   final Movies? trendingMovies;
 
-  CardSwiper({required this.trendingMovies});
+  CardSwiper({required this.trendingMovies, required this.isTextFieldEmpty});
```
You missed a trailing comma here

Kevin-Guillen replied:
added

MarcosCGlobant commented on lib/src/widgets/card_swiper.dart:
```diff
@@ -4,9 +4,11 @@ import '../model/movies.dart';
   final Movies? trendingMovies;
 
-  CardSwiper({required this.trendingMovies});
+  CardSwiper({required this.trendingMovies, required this.isTextFieldEmpty});
```
missed a trailing comma here

MarcosCGlobant commented on lib/src/widgets/card_swiper.dart:
```diff
@@ -41,6 +43,7 @@ class CardSwiper extends StatelessWidget {
                         fit: BoxFit.fill,
                       );
                     },
+                    control: SwiperControl(color: Colors.purple),
```
missed a trailing comma here

Kevin-Guillen replied:
added

MarcosCGlobant commented on lib/src/widgets/card_swiper.dart:
```diff
@@ -19,7 +21,7 @@ class CardSwiper extends StatelessWidget {
               ),
               child: new Text(
-                'Trending Movies',
+                isTextFieldEmpty ? "Trending Movies" : "Search results",
```
these messages should be on the constants file

Kevin-Guillen replied:
done

marianosalvetti reviewed (approved):
I like this PR, it's fine if you fix the minors comments, it's ready to merge.

FranciscoGaraventa reviewed (approved):
LGTM

MarcosCGlobant reviewed (approved):
LGTM
